pkg/registration/clientcert/certificate.go

```diff
@@ -147,13 +147,19 @@ func getCertValidityPeriod(secret *corev1.Secret) (*time.Time, *time.Time, error
 
 // BuildKubeconfig builds a kubeconfig based on a rest config template with a cert/key pair
 func BuildKubeconfig(clientConfig *restclient.Config, certPath, keyPath string) clientcmdapi.Config {
```

Comment:
why we still need this func? or I think we can just have BuildKubeconfig to also use proxyURL as an input, or the we should use cliemtcmdapi.config instead of rest.Config as the input.

Reply:
The signature of BuildKubeconfig has been updated as suggested.